austrakka/components/project/dataset/funcs.py

```diff
@@ -0,0 +1,127 @@
+import os
+import time
+
+import pandas as pd
+from loguru import logger
+from austrakka.utils.api import api_post, api_get
+from austrakka.utils.fs import get_hash
+from austrakka.utils.helpers.output import call_get_and_print_table_on_state_change
+from austrakka.utils.helpers.output import call_get_and_print_dataset_status
+from austrakka.utils.helpers.upload import upload_multipart_tracking_token
+from austrakka.utils.misc import logger_wraps
+from austrakka.utils.output import print_table
+from austrakka.utils.paths import PROJECT_PATH
+
+DATASET_UPLOAD_PATH = 'dataset'
+DATASET_ACK_PATH = 'acknowledge'
+DATASET_TRACK_PATH = 'dataset-progress'
+DATASET_TRACK_DETAILED_PATH = 'dataset-progress-details'
+
+
+@logger_wraps()
+def add_dataset(
+        filepath: str,
+        label: str,
+        abbrev: str):
+
+    path = "/".join([PROJECT_PATH, abbrev, DATASET_UPLOAD_PATH])
+    filename = os.path.basename(filepath)
+    custom_headers = {
+        'analysis-label': label,
+        'filename': filename,
+    }
+
+    file_hash = get_hash(filepath)
+    with open(filepath, 'rb') as file_content:
+        files = [('files[]', (filename, file_content))]
+        tracking_token = upload_multipart_tracking_token(path=path,
+                                                         files=files,
+                                                         file_hash=file_hash,
+                                                         custom_headers=custom_headers)
+    logger.info('Acknowledging...')
+    path_ack = "/".join([PROJECT_PATH, abbrev, DATASET_ACK_PATH, tracking_token])
+    return api_post(
+        path=path_ack,
+    )
+
+
+@logger_wraps()
+def track_dataset(
+        abbrev:str,
+        tracking_token: str,
+        detailed: bool,
+        out_format: str):
+    path = "/".join([PROJECT_PATH,
+                     abbrev,
+                     DATASET_TRACK_DETAILED_PATH if detailed else DATASET_TRACK_PATH,
+                     tracking_token])
+    if detailed:
+        response = api_get(path)
+        data = response['data'] if ('data' in response) else response
+        if not data:
+            logger.info("No JobFeedbacks available")
+            return
```

Comment:
Under what circumstances will there be no JobFeedbacks? No recent uploads? No unfinished data ingestions?

I'm thinking of avoiding "job" in the message, but I'm not sure what the message really signifies in practice.

Reply:
If the dataset as been uploaded but not acknowledged (using add-async) command. Feedback starts only from the validation stage onward.